src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -313,20 +313,45 @@ public function dump(array $options = [])
 
 use Symfony\Component\DependencyInjection\Dumper\Preloader;
 
-require $autoloadFile;
-require __DIR__.'/Container{$hash}/{$options['class']}.php';
+require $vendorDir.'/ClassLoader.php';
+\$loader = new \Composer\Autoload\ClassLoader();
+
+\$map = require $vendorDir.'/autoload_namespaces.php';
+foreach (\$map as \$namespace => \$path) {
```

Comment:
why reconfiguring a ClassLoader instead manually here instead of using `vendor/autoload.php` ? And why not using the `autoload_static.php` file, which is the one used by composer on PHP 5.6+, which relies on immutable arrays in shared memory ?

Reply:
Requiring `vendor/autoload.php` causes issues with preloading since it loads all the `autoload_files` immediately and currently causes a segfault in some cases. So we need to create an autoloader without actually loading any files, in order to look up the paths for the files.

And this is a compiled file which is used for preloading, so I'm not really sure if using immutable arrays in shared memory would really make any difference here